src/main/java/randoop/generation/OrienteeringSelection.java

```diff
@@ -0,0 +1,252 @@
+   * retrieve their respective execution times from the eSeq object.
+   *
+   * <p>If the statement executed normally but has a measured time of 0 nanoseconds, we assign it an
+   * execution time of 1 nanosecond. This is to prevent division by zero. An example of a statement
+   * with a measured execution time of zero is assignment statements of primitive types with a
+   * constant value, such as {@code byte byte0 = (byte)1;}.
+   *
+   * <p>We do not update the execution time of an input sequence once it has been assigned. This is
+   * because we do not believe that a single input sequence's execution time will change drastically
+   * between different runs. This is a simplification of GRT's description of Orienteering, which
+   * does differentiate execution times of a given sequence between multiple runs.
+   *
+   * @param inputSequences the sequences that were chosen as the input to the method under test for
+   *     creating {@code eSeq}; each one is a subsequence of {@code eSeq}
+   * @param eSeq the recently executed sequence which is new and unique, and has just been executed.
+   *     It contains its overall execution time. It also contains, for each statement, an
+   *     ExecutionOutcome that gives the statement's execution time.
+   */
+  @Override
+  public void createdExecutableSequenceFromInputs(
+      List<Sequence> inputSequences, ExecutableSequence eSeq) {
+    Map<Statement, Long> statementExecTimeMap = new HashMap<>();
+
+    // Iterate through the executable sequence and populate the map, mapping from statement to
+    // execution time. There will be an execution outcome for each statement since we have
+    // the invariant in ExecutableSequence that sequence.size() == executionResults.size().
+    for (int i = 0; i < eSeq.size(); i++) {
+      Statement statement = eSeq.sequence.getStatement(i);
+      ExecutionOutcome executionOutcome = eSeq.getResult(i);
+
+      if (executionOutcome instanceof NormalExecution) {
+        Long statementExecTime = executionOutcome.getExecutionTime();
+        // If the statement executed normally but has a measured time of 0 nanoseconds, we assign it
+        // an execution time of 1 nanosecond. This is to prevent division by zero when computing
+        // weights which uses the statement's execution time as part of a product in the
+        // denominator.
+        if (statementExecTime == 0) {
+          statementExecTime = 1L;
+        }
+        statementExecTimeMap.put(statement, statementExecTime);
+      } else {
+        // If the statement did not execute normally, we assign it a measured time of -1
+        // nanoseconds.  We will check for this below and subsequently skip computing the execution
+        // time of any sequence that contains a statement that did not execute normally.
+        statementExecTimeMap.put(statement, -1L);
+      }
+    }
+
+    // We compute the execution time of each input sequence for the given eSeq.
```

Comment:
Wasn't the execution time of each input sequence computed when it was first created (when it was `eSeq`)?  I don't see why recomputation is necessary.

More generally, I don't understand the algorithm used in this method.  I presume its goal is to compute the execution time of `eSeq`.  That could be done in two ways:
 * as the sum of the execution times of the input sequences, plus the execution time of the final method call
 * as the sum of the execution times of all the statements in the sequence
Either approach works, and either one is sufficient.  However, this method does some computations over all the statements, and some computations over the input sequences.  I'm not sure why.  I may be missing something.  Can you explain, or clean up the code?